modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/portlet.js

```diff
@@ -152,6 +177,33 @@
 		},
 	};
 
+	function destroyPortlet(options) {
+
+		// NOTE: Will have to vet all the current listeners to see what
+		// `options` they might care about. A cursory examination suggests that
+		// it is probably only ever (or mostly) `portletId`.
+
+		const {portletId} = options;
+
+		let activeIds = State.readKey(ACTIVE_PORTLET_IDS);
+
+		if (!(portletId in activeIds)) {
+			console.warn(
+				`destroyPortlet(): ${portletId} is not in the set of active portlet IDs`
+			);
+		}
+
+		activeIds = {...activeIds};
+
+		delete activeIds[portletId];
+
+		State.writeKey(ACTIVE_PORTLET_IDS, activeIds);
+
+		// Backwards compatibility.
+
+		Liferay.fire('destroyPortlet', options);
```

Comment:
Presumably we're going to have to keep this here for customers, even if we migrate all of our own internal usages before 7.4 goes final (ie. deprecate in 7.4, remove in 7.5).